internal/resource/http.go

```diff
@@ -105,30 +105,40 @@ func (f *Fetcher) UpdateHttpTimeoutsAndCAs(timeouts types.Timeouts, cas []types.
 	}
 
 	for _, ca := range cas {
-		cablob, err := f.getCABlob(ca)
+		sources, err := ca.GetSources()
 		if err != nil {
 			return err
 		}
-		if err := f.parseCABundle(cablob, ca, pool); err != nil {
-			f.Logger.Err("Unable to parse CA bundle: %s", err)
-			return err
+		if len(sources) == 0 {
+			f.Logger.Crit("invalid CA: %v", ignerrors.ErrSourceRequired)
+			return ignerrors.ErrSourceRequired
+		}
+		for _, src := range sources {
+			cablob, err := f.getCABlob(ca, string(src))
+			if err != nil {
+				return err
+			}
+			if err := f.parseCABundle(cablob, ca, string(src), pool); err != nil {
+				f.Logger.Err("Unable to parse CA bundle: %s", err)
+				return err
+			}
```

Comment:
Isn't this adding the CA bundles from each source rather than fetching from all of them and only selecting from the fastest one?

Reply:
this part i haven't finished yet , but all the rest works (at least in blackbox tests)